### PR TITLE
fix(gatsby): filter out not applicable flags (isCi / command) when constructing flags message

### PR DESCRIPTION
**Pull request by pieh**
## Description

Noticed weird CLI messages during work on https://github.com/gatsbyjs/gatsby/pull/30857 where message wouldn't be displayed correctly due https://github.com/gatsbyjs/gatsby/blob/06687500de521d8ce8710a96ec817f87430f80a0/packages/gatsby/src/utils/handle-flags.ts#L145-L155

not being satisfied and not displaying which flags are active because `lockedInFlags` would contain flags for develop which were not actually enabled.

This also make it so only applicable flags (so flags that satisfy CI condition and command condition) are listed in messages (i.e. `DEV_SSR` won't be printed for builds)

### CODE REVIEW

File: packages/gatsby/src/utils/__tests__/handle-flags.ts [hunk 1]
```diff
@@ -332,6 +330,60 @@ describe(`handle flags`, () => {
...
+              "experimental": false,
+              "name": "SOME_FLAG",
+              "telemetryId": "test",
+              "testFitness": [Function],
+              "umbrellaIssue": "test",
+            },
+          ],
+          "message": "The following flags are active:
+        - SOME_FLAG · (Umbrella Issue (test)) · test
+        ",
```
Comment (pieh) [inline packages/gatsby/src/utils/__tests__/handle-flags.ts:381]: `message` in this test would be empty string without changes to `handle-flags.ts`

File: packages/gatsby/src/utils/handle-flags.ts [hunk 2]
```diff
@@ -69,6 +69,15 @@ const handleFlags = (
   const lockedInFlags = new Map<string, IFlag>()
   const lockedInFlagsThatAreInConfig = new Map<string, IFlag>()
   availableFlags.forEach(flag => {
+    const isForCommand =
+      flag.command === `all` || flag.command === executingCommand
+    // If we're in CI, filter out any flags that don't want to be enabled in CI
+    const isForCi = isCI() ? flag.noCI !== true : true
```
Comment (vladar) [inline packages/gatsby/src/utils/handle-flags.ts:75]: Nit: a bit hard to follow. So if `isCI()` is `false` then `isForCi` is always `true`?

Comment (pieh) [reply packages/gatsby/src/utils/handle-flags.ts:75]: True

TBH this was copypasta from few lines below https://github.com/gatsbyjs/gatsby/blob/06687500de521d8ce8710a96ec817f87430f80a0/packages/gatsby/src/utils/handle-flags.ts#L96-L100

But I'll think of making it easier to read

Comment (pieh) [reply packages/gatsby/src/utils/handle-flags.ts]: I tried to make it easier to read - not sure if I succeeded - https://github.com/gatsbyjs/gatsby/pull/30977/commits/35086b3d2b236230340881dc4546106d3eae9269

File: packages/gatsby/src/utils/__tests__/__snapshots__/handle-flags.ts.snap [hunk 3]
```diff
@@ -50,11 +50,9 @@ The following flags were automatically enabled on your site:
 - PARTIAL_RELEASE · (Umbrella Issue (test)) · test
 - PARTIAL_RELEASE_ONLY_NEW_LODASH · (Umbrella Issue (test)) · test
 
-There are 5 other flags available that you might be interested in:
+There are 3 other flags available that you might be interested in:
 - FAST_DEV · Enable all experiments aimed at improving develop server start time
 - DEV_SSR · (Umbrella Issue (https://github.com/gatsbyjs/gatsby/discussions/28138)) · SSR pages on full reloads during develop. Helps you detect SSR bugs and fix them without needing to do full builds.
-- QUERY_ON_DEMAND · (Umbrella Issue (https://github.com/gatsbyjs/gatsby/discussions/27620)) · Only run queries when needed instead of running all queries upfront. Speeds starting the develop server.
```
Comment (pieh) [inline packages/gatsby/src/utils/__tests__/__snapshots__/handle-flags.ts.snap:56]: this is removed because `QUERY_ON_DEMAND` is not available in CI (side note - because it's locked in we probably shouldn't display it for other reasons, but that seems separate)

File: packages/gatsby/src/utils/__tests__/__snapshots__/handle-flags.ts.snap [hunk 4]
```diff
@@ -50,11 +50,9 @@ The following flags were automatically enabled on your site:
...
-- ONLY_BUILDS · (Umbrella Issue (test)) · test
```
Comment (pieh) [inline packages/gatsby/src/utils/__tests__/__snapshots__/handle-flags.ts.snap:57]: This is removed because we run test for `develop` command and this flag is only available for builds

**Review by vladar — APPROVED** [review]
Thank you 👍
